src/components/player/Player.tsx

```diff
@@ -230,54 +222,52 @@ export function Player({ player }: { player: any }) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [currentSong, player, progress, state]);
 
-  // Error Event Effect
+  // Error handling: try to reload
   useEffect(() => {
-    if (player && hasError && currentSong) {
-      // Initialize retry count
-      if (!retryCounts[currentSong.video_id])
-        retryCounts[currentSong.video_id] = 0;
-
-      // Allow up to 3 retries
-      if (retryCounts[currentSong.video_id] <= 2) {
-        if (getID(player.getVideoUrl()) !== currentSong.video_id) return;
-        setTimeout(() => {
-          console.log(
-            `[Player] Retrying ${currentSong.name} - attempt #${
-              retryCounts[currentSong.video_id]
-            }/3`
-          );
-          player.loadVideoById(currentSong.video_id, currentSong.start);
-          setError(false);
-        }, 2000);
-        retryCounts[currentSong.video_id] += 1;
-        return;
-      }
-
-      console.log(
-        "[PLAYER] SKIPPING____ DUE TO VIDEO PLAYBACK FAILURE (maybe the video is blocked in your country)"
-      );
-      toast({
-        position: "top-right",
-        status: "warning",
-        title: `The Song: ${currentSong?.name} is not playable. Skipping it.`,
-        isClosable: true,
-      });
-      next({ count: 1, userSkipped: false, hasError: true });
-      setError(false);
+    const MAX_RETRIES = 3;
+    if (!(player && hasError && currentSong)) return;
+
+    if (!retryCounts[currentSong.video_id])
+      retryCounts[currentSong.video_id] = 0;
+
+    if (retryCounts[currentSong.video_id] < MAX_RETRIES) {
+      // if (getID(player.getVideoUrl()) !== currentSong.video_id) return;
```

Comment:
Not sure how to do async sequential timeouts inside of useEffect so just commented out that line for now.

Reply:
Can you add a `NOTE:` comment in the code as well for future reference?